src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -1446,6 +1448,33 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
 
         $loader->load('messenger.xml');
 
+        $builtBuses = array();
+        foreach ($config['buses'] as $name => $bus) {
+            $middlewares = array_map(function (array $middleware) use ($container) {
+                if ($container->has($messengerMiddlewareId = 'messenger.middleware.'.$middleware['service'])) {
+                    $middleware['service'] = $messengerMiddlewareId;
+                }
+
+                return new Reference($middleware['service']);
+            }, $bus['middlewares']);
+
+            $container->setDefinition($busId = 'messenger.bus.'.$name, (new Definition(MessageBus::class, array($middlewares)))->addTag('messenger.bus'));
+
+            if ($name === $config['default_bus']) {
```

Comment:
dx-wise i'd expect an exception in case of an unknown bus is configured to be the default